src/xas_tdp_integrals.F

```diff
@@ -547,11 +607,10 @@ SUBROUTINE fill_pqX_tensor(pq_X, ab_nl, ac_nl, basis_set_list_a, basis_set_list_
                   END IF
 
                   ALLOCATE (work(nsgfa(iset), nsgfb(jset), nsgfc(kset)))
-                  !zeroing work is superfluous here (work = 0.0_dp)
 
-                  CALL abc_contract(work, sabc, &
-                                    sphi_a(:, sgfa:), sphi_b(:, sgfb:), sphi_c(:, sgfc:), &
-                                    ncoa, ncob, ncoc, nsgfa(iset), nsgfb(jset), nsgfc(kset))
+                  CALL libxsmm_abc_contract(work, sabc, tspa(iset, ikind)%array, spb(jset, jkind)%array, &
```

Comment:
`libxsmm_abc_contract` is a no-op without libxsmm, so this results in no contraction then? Or did I miss something?

Reply:
There is a fallback on BLAS in the case where libxsmm is not linked, which yields the same result, just not as fast.

Reply:
ok, I missed the `#else` in `libxsmm_abc_contract`. But this kind of stuff is just one huge anti-pattern, we have to come up with something better at some point. Maybe an optional attribute `preferred_driver` for routines which can use different backends to select the preferred driver for the operation in this call? Well, but not your issue here.